tests/testsuite/hints.rs

```diff
@@ -49,11 +49,197 @@ fn unknown_hints_warn() {
         .build();
     p.cargo("check -v")
         .with_stderr_data(str![[r#"
-[WARNING] unused manifest key: hints
+[WARNING] unused manifest key: hints.this-is-an-unknown-hint
 [CHECKING] foo v0.0.1 ([ROOT]/foo)
 [RUNNING] `rustc --crate-name foo [..]`
 [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
 
 "#]])
         .run();
 }
+
+#[cargo_test]
+fn hints_mostly_unused_warn_without_gate() {
```

Comment:
I had meant for all of these tests to be added in the prior test commit

Reply:
Alright, I've now pre-added versions of all the tests, and then evolved them to reflect the desired behavior in the commit adding that behavior.